### DISCUSSION TITLE
How to add global variable as input

### DISCUSSION THREAD

**Discussion by zhaogw0809** [danielegrattarola/spektral#213] [Q&A]
Hi,

I am a student trying to learn spektral following your example custom_dataset.py. The difference is that my GCN input X_in not only requires to include each node's features which is a matrix like your example, but also should contains some global variables which is a value like graph depth. So, I just wondering how to combine these two parts together or make them as a whole, then put them into  input layer. Could you give me some suggestions?

Thanks anyway

**Comment by danielegrattarola** [comment]
Hi, 

it depends on how you want to process this global information in your network, and what task you are trying to solve. 

Is it a node-level or graph-level learning task?
Do you want the global information to be taken into account by the message passing layers?

Cheers

**Reply by zhaogw0809** [reply]
Thanks for your reply.

It is a graph-level task and required the global information to pass layers. Any suggestions?

**Reply by danielegrattarola** [reply]
You have a couple of alternatives: 

1. Concatenate the global attribute to all node features (you'll need to broadcast it so that it has shape (N, attr_size))
2. Concatenate the global attribute to the vector that you obtain after global pooling

Both should be valid solutions.
Cheers

**Reply by zhaogw0809** [reply]
Really appreciate your advice. Have a nice day, sir!
